libtenzir/builtins/formats/lines.cpp

```diff
@@ -83,6 +93,13 @@ class lines_parser final : public plugin_parser {
         if (line->empty() and skip_empty) {
           continue;
         }
+        // TODO: Or should this be a warning?
+        if (not arrow::util::ValidateUTF8(*line)) {
+          diagnostic::error("got invalid UTF-8")
+            .primary(self)
+            .emit(ctrl.diagnostics());
+          co_return;
+        }
```

Comment:
Not sure about this. Normally we want to continue, but this strongly hints that the incoming stream is binary as a whole, so the occasional line being valid UTF-8 is just luck.

Reply:
I'd treat this similar to `file_contents`: Add a `binary=true` option that, when set, changes the line to be of type `blob` and removes this check.

Reply:
NB: We currently use a mix of simdjson's and Arrow's UTF8 validation in the code base. Do we know which one is faster? Intuitively I'd always go for simdjson's.

Reply:
I can add this option and agree that we should have it, but the question was more about whether we should error when `binary=false` or rather warn.

Reply:
I prefer warn+drop

Reply:
Added `binary=true` with your desired semantics.

> NB: We currently use a mix of simdjson's and Arrow's UTF8 validation in the code base. Do we know which one is faster? Intuitively I'd always go for simdjson's.

Not sure, but also probably not something to find out for this PR.